crates/tabby-index/src/indexer.rs

```diff
@@ -72,21 +72,39 @@ impl<T: ToIndexId> TantivyDocBuilder<T> {
         let now = tantivy::time::OffsetDateTime::now_utc();
         let updated_at = tantivy::DateTime::from_utc(now);
 
-        let doc = doc! {
-            schema.field_id => id,
-            schema.field_source_id => source_id,
-            schema.field_corpus => self.corpus,
-            schema.field_attributes => self.builder.build_attributes(&document).await,
-            schema.field_updated_at => updated_at,
-        };
-
         let cloned_id = id.clone();
+        let doc_id = id.clone();
+        let doc_attributes = self.builder.build_attributes(&document).await;
         let s = stream! {
-            yield tokio::spawn(async move { Some(doc) });
+            let mut failed_count: u64 = 0;
+            for await chunk_doc in self.build_chunks(cloned_id, source_id.clone(), updated_at, document).await {
+                match chunk_doc.await {
+                    Ok((Some(doc), ok)) => {
+                        if !ok {
+                            failed_count += 1;
+                        }
+                        yield tokio::spawn(async move { Some(doc) });
```

Comment:
do it still need to yield when `!ok`?

Reply:
yes, the doc itself still matters and could be used to search in tantivy, the `!ok` only indicates the embedding is failed calculated.